unified_planning/model/action.py

```diff
@@ -764,3 +764,228 @@ def __repr__(self) -> str:
+        self._simulated_effect: Optional[up.model.effect.SimulatedEffect] = None
+        # fluent assigned is the mapping of the fluent to it's value if it is an unconditional assignment
+        self._fluents_assigned: Dict[
+            "up.model.fnode.FNode", "up.model.fnode.FNode"
+        ] = {}
+        # fluent_inc_dec is the set of the fluents that have an unconditional increase or decrease
+        self._fluents_inc_dec: Set["up.model.fnode.FNode"] = set()
+
+    def __repr__(self) -> str:
+        s = []
+        s.append(f"process {self.name}")
+        first = True
+        for p in self.parameters:
+            if first:
+                s.append("(")
+                first = False
+            else:
+                s.append(", ")
+            s.append(str(p))
+        if not first:
+            s.append(")")
+        s.append(" {\n")
+        s.append("    preconditions = [\n")
+        for c in self.preconditions:
+            s.append(f"      {str(c)}\n")
+        s.append("    ]\n")
+        s.append("    effects = [\n")
+        for e in self.effects:
+            s.append(f"      {str(e)}\n")
+        s.append("    ]\n")
+        s.append("  }")
+        return "".join(s)
+
+    def __eq__(self, oth: object) -> bool:
+        if isinstance(oth, Process):
+            cond = (
+                self._environment == oth._environment
+                and self._name == oth._name
+                and self._parameters == oth._parameters
+            )
+            return (
+                cond
+                and set(self._preconditions) == set(oth._preconditions)
+                and set(self._effects) == set(oth._effects)
+                and self._simulated_effect == oth._simulated_effect
+            )
+        else:
+            return False
+
+    def __hash__(self) -> int:
+        res = hash(self._name)
+        for ap in self._parameters.items():
+            res += hash(ap)
+        for p in self._preconditions:
+            res += hash(p)
+        for e in self._effects:
+            res += hash(e)
+        res += hash(self._simulated_effect)
+        return res
+
+    def clone(self):
+        new_params = OrderedDict(
+            (param_name, param.type) for param_name, param in self._parameters.items()
+        )
+        new_instantaneous_action = Process(self._name, new_params, self._environment)
+        new_instantaneous_action._preconditions = self._preconditions[:]
+        new_instantaneous_action._effects = [e.clone() for e in self._effects]
+        new_instantaneous_action._fluents_assigned = self._fluents_assigned.copy()
+        new_instantaneous_action._fluents_inc_dec = self._fluents_inc_dec.copy()
+        new_instantaneous_action._simulated_effect = self._simulated_effect
+        return new_instantaneous_action
+
+    @property
+    def preconditions(self) -> List["up.model.fnode.FNode"]:
+        """Returns the `list` of the `Action` `preconditions`."""
```

Comment:
All the comments in this class are copy-pasted, we need to replace `Action` with `Process`

Reply:
Yep